### PR TITLE
ethdb: golint fixes and comments added from leveldb docs

### PR DESCRIPTION
**Pull request by kielbarry**

### CODE REVIEW

File: ethdb/database.go [hunk 1]
```diff
@@ -150,6 +154,7 @@ func (db *LDBDatabase) Close() {
 	}
 }
 
+// LDB returns the memory address for leveldb.DB.
```
Comment (karalabe) [inline ethdb/database.go]: `// LDB returns the internal leveldb.DB instance.`

File: ethdb/database.go [hunk 2]
```diff
@@ -356,6 +361,7 @@ func (db *LDBDatabase) meter(refresh time.Duration) {
 	}
 }
 
+// NewBatch populates the ldbBatch for read and writes to the db.
```
Comment (karalabe) [inline ethdb/database.go]:
```
// NewBatch creates a memory buffer to group together database writes and flush
// them out in one go. Batches are not atomic, just an performance optimization.
```

File: ethdb/interface.go [hunk 3]
```diff
@@ -16,7 +16,7 @@
 
 package ethdb
 
-// Code using batches should try to add this much data to the batch.
+// IdealBatchSize is the amount of data vode using batches should try to add to the batch.
```
Comment (karalabe) [inline ethdb/interface.go]: `// IdealBatchSize is the amount of data batches should accumulate before flushing.`

File: ethdb/memory_database.go [hunk 4]
```diff
@@ -26,23 +26,28 @@ import (
...
+
+// MemDatabase imitates the key-value store levelDB for the test memory database.
 type MemDatabase struct {
 	db   map[string][]byte
 	lock sync.RWMutex
 }
 
-func NewMemDatabase() *MemDatabase {
+// NewMemDatabase inits a mock levelDB instance with a map.
+func NewMemDatabase() (*MemDatabase, error) {
```
Comment (karalabe) [inline ethdb/memory_database.go]: Please rebase your code on top of master, this signature changed.

File: ethdb/memory_database.go [hunk 5]
```diff
@@ -26,23 +26,28 @@ import (
 /*
  * This is a test memory database. Do not use for any production it does not get persisted
  */
+
+// MemDatabase imitates the key-value store levelDB for the test memory database.
```
Comment (karalabe) [inline ethdb/memory_database.go]: Please delete the top comment altogether ("This is a test memory database ..."). As for the main doc, let's make it a bit nicer (it's not only used for testing for a long time now).

`
// MemDatabase is an in-memory key-value store.
`

File: ethdb/memory_database.go [hunk 6]
```diff
@@ -26,23 +26,28 @@ import (
...
 type MemDatabase struct {
 	db   map[string][]byte
 	lock sync.RWMutex
 }
 
-func NewMemDatabase() *MemDatabase {
+// NewMemDatabase inits a mock levelDB instance with a map.
```
Comment (karalabe) [inline ethdb/memory_database.go]: MemDatabase is not a leveldb mock, it's a memory key/value store.

```
// NewMemDatabase creates an in-memory key-value store.
```

File: ethdb/memory_database.go [hunk 7]
```diff
@@ -26,23 +26,28 @@ import (
...
+func NewMemDatabase() (*MemDatabase, error) {
 	return &MemDatabase{
 		db: make(map[string][]byte),
 	}
 }
 
-func NewMemDatabaseWithCap(size int) *MemDatabase {
+// NewMemDatabaseWithCap inits a mock levelDB instance with a map and sets a maximum size..
```
Comment (karalabe) [inline ethdb/memory_database.go]: MemDatabase is not a leveldb mock, it's a memory key/value store.

```
// NewMemDatabaseWithCap creates an in-memory key-value store with a specific
// starting capacity item.
```

File: ethdb/memory_database.go [hunk 8]
```diff
@@ -59,6 +65,7 @@ func (db *MemDatabase) Has(key []byte) (bool, error) {
 	return ok, nil
 }
 
+// Get returns an error if the given key is not found.
```
Comment (karalabe) [inline ethdb/memory_database.go]: `// Get retrieves the value associated with a given key.`

File: ethdb/memory_database.go [hunk 9]
```diff
@@ -69,6 +76,7 @@ func (db *MemDatabase) Get(key []byte) ([]byte, error) {
 	return nil, errors.New("not found")
 }
 
+// Keys returns a list of all keys in db.db.
```
Comment (karalabe) [inline ethdb/memory_database.go]: `// Keys returns a list of all the keys present in the database.`

File: ethdb/memory_database.go [hunk 10]
```diff
@@ -80,6 +88,7 @@ func (db *MemDatabase) Keys() [][]byte {
 	return keys
 }
 
+// Delete deletes the key from db.db.
```
Comment (karalabe) [inline ethdb/memory_database.go]: ` // Delete removes the specified entry from the database.`

File: ethdb/memory_database.go [hunk 11]
```diff
@@ -88,12 +97,15 @@ func (db *MemDatabase) Delete(key []byte) error {
 	return nil
 }
 
+// Close performs no operation but imitates invocation of levelDB.Close().
```
Comment (karalabe) [inline ethdb/memory_database.go]: `// Close implements the closer interface. For the memory database, this is a noop.`

File: ethdb/memory_database.go [hunk 12]
```diff
@@ -88,12 +97,15 @@ func (db *MemDatabase) Delete(key []byte) error {
...
 func (db *MemDatabase) Close() {}
 
+// NewBatch sets memBatch.db equal to the receiver.
```
Comment (karalabe) [inline ethdb/memory_database.go]:
```
// NewBatch creates a memory buffer to group together database writes and flush
// them out in one go. Batches are not atomic, just an performance optimization.
```

File: ethdb/memory_database.go [hunk 13]
```diff
@@ -88,12 +97,15 @@ func (db *MemDatabase) Delete(key []byte) error {
...
 func (db *MemDatabase) NewBatch() Batch {
 	return &memBatch{db: db}
 }
 
+// Len returns the number of keys in db.db.
```
Comment (karalabe) [inline ethdb/memory_database.go]: `// Len returns the number of entries in the database.`

File: ethdb/database.go (hunk 1 above)
Comment (kielbarry) [reply ethdb/database.go]: Thank you for all the additions, corrections, and feedback on this module, Peter. I'll be updating the PR tomorrow.

**Comment by holiman** [comment]
@kielbarry sorry it's taken so long - could you fix the conflict and we can get this merged? Thanks

**Comment by fjl** [comment]
Sorry, this is now stale, closing. Documentation issues have been adressed in #19021.

**Pull request closed without merging**